### PR TITLE
feat: add routing parameters to widget config

### PR DESCRIPTION
**Pull request by mikasackermn**
# Summary

We needed to add `avoidNativeFee` and `maxLength` to features, but after speaking with @RanGojo and @yeager-eren, we concluded that we should create a new parameter called `routing`. This parameter will contain `avoidNativeFee`, `maxLength`, and `experimental`.

### CODE REVIEW

File: widget/embedded/src/types/config.ts [hunk 1]
```diff
@@ -143,8 +140,30 @@ export type Features = Partial<
...
   >
+>;
+
+/**
+ * `Routing`
+ *
+ * @property {'disabled' | 'enabled'} [avoidNativeFee]
+ * - The enablement state for the avoid native fee feature. Optional property.
+ *
+ * @property {number} [maxLength]
```
Comment (nikaaru) [inline widget/embedded/src/types/config.ts:151]: I think we can express these comments more clearly
Or, for example, take it from the document here:

https://api-docs.rango.exchange/reference/getbestroutes

File: widget/embedded/src/types/config.ts [hunk 2]
```diff
@@ -143,8 +140,30 @@ export type Features = Partial<
     | 'customTokens',
     'visible' | 'hidden'
   >
+>;
+
+/**
+ * `Routing`
+ *
+ * @property {'disabled' | 'enabled'} [avoidNativeFee]
+ * - If you want to exclude Swappers that have native tokens as a fee, for example when calling it from an AA account, you could set this parameter to enabled.
```
Comment (RanGojo) [inline widget/embedded/src/types/config.ts:149]:
```suggestion
 * - If you want to exclude swappers that have native tokens as fee, for example when your call widget using an AA account, you could set this parameter to true.
```

File: widget/embedded/src/types/config.ts [hunk 3]
```diff
@@ -143,8 +140,30 @@ export type Features = Partial<
...
+ *
+ * @property {number} [maxLength]
+ * - If you want to set the maximum acceptable length of a path, you could set this parameter to a numeric value.
```
Comment (RanGojo) [inline widget/embedded/src/types/config.ts:152]:
```suggestion
 * - If you want to limit the maximum acceptable length of a route, you could set this parameter.
```

File: widget/embedded/src/types/config.ts [hunk 4]
```diff
@@ -143,8 +140,30 @@ export type Features = Partial<
...
+ *
+ * @property {'disabled' | 'enabled'} [experimental]
+ * - The enablement state for the experimental route feature. Optional property.
```
Comment (RanGojo) [inline widget/embedded/src/types/config.ts:155]:
```suggestion
 * - Use this property to enable experimental routes. This settings shouldn't be used in your production environment.
```

File: widget/embedded/src/types/config.ts [hunk 5]
```diff
@@ -192,22 +211,24 @@ export type TrezorManifest = {
...
  *   - 'language': Visibility state for the language.
  *   - 'customTokens': Visibility state for the custom tokens.
- *   - 'experimentalRoute': Enablement state for the experimental route.
- *
+ * @property {Routing} routing - An optional object for configuring the enablement or numeric of various features.
+ *   Keys include:
+ *   - 'maxLength': Numeric state for the max length.
+ *   - 'avoidNativeFee': Enablement state for the avoid native fee.
+ *   - 'experimental': Enablement state for the experimental route.
+ *   - 'enableCentralizedSwappers': Enablement state for the centralized swappers.
```
Comment (RanGojo) [inline widget/embedded/src/types/config.ts:227]: Please also update here.

File: widget/embedded/src/types/config.ts (hunk 2 above)
Comment (mikasackermn) [reply widget/embedded/src/types/config.ts]: fixed

File: widget/embedded/src/types/config.ts (hunk 3 above)
Comment (mikasackermn) [reply widget/embedded/src/types/config.ts]: fixed

File: widget/embedded/src/types/config.ts (hunk 5 above)
Comment (mikasackermn) [reply widget/embedded/src/types/config.ts]: fixed

File: widget/embedded/src/types/config.ts (hunk 1 above)
Comment (mikasackermn) [reply widget/embedded/src/types/config.ts:151]: fixed

**Review by nikaaru — APPROVED** [review]
LGTM

**Pull request merged**